src/haz3lcore/statics/Statics.re

```diff
@@ -636,3 +636,82 @@ let collect_errors = (map: Map.t): list((Id.t, Info.error)) =>
     map,
     [],
   );
+
+/*
+ * Check whether the argument of a function under the new annotation is of correct format,
+ * and output a list of input types
+ */
+let caf_argtrans =
+    (args: list(UPat.t)): option((list(UPat.t), list(UTyp.t))) =>
+  List.fold_right(
```

Comment:
don't need to fold here, you can just recursively synthesize a type for the args pattern since that will be a product type that we can use directly as the input type